lisa/wlgen/workload.py

```diff
@@ -72,24 +72,22 @@ def run(self, cpus=None, cgroup=None, background=False, as_root=False, value=42)
     required_tools = ['taskset']
     """
     The tools required to execute the workload. See
-    :meth:`lisa.env.TestEnv.install_tools`.
+    :meth:`lisa.target.Target.install_tools`.
     """
 
-    def __init__(self, te, name, res_dir=None):
-        self.te = te
+    def __init__(self, target, name, res_dir=None):
+        self.target = target
         self.name = name
-
-        # XXX: move this to run() instead?
-        if not res_dir:
-            res_dir = self.te.get_res_dir(name)
-
-        self.res_dir = res_dir
-
         self.command = None
         self.output = ""
-        self.run_dir = self.te.target.working_directory
+        self.run_dir = self.target.working_directory
+
+        res_dir = res_dir if res_dir else target.get_res_dir(
+            name='{}-{}'.format(self.__class__.__qualname__, name)
```

Comment:
We could probably drop the "name" requirement at this point, but seeing how big this PR is I'll do that separately.